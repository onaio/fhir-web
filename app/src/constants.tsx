export const LOGOUT_REDIRECTION_DELAY = 1000;
export const REACT_CALLBACK_PATH = '/oauth/callback/:id';
export const BACKEND_CALLBACK_PATH = '/fe/oauth/callback/:id';

// URLs
export const URL_ADMIN = '/admin';
export const URL_MISSIONS = '/missions';
export const URL_EXPRESS_LOGIN = '/login';
export const URL_REACT_LOGIN = '/login';
export const URL_LOGOUT = '/logout';
export const URL_HOME = '/';

export const URL_USER = `${URL_ADMIN}/users/list`;
export const URL_USER_GROUPS = `${URL_ADMIN}/users/groups`;
export const URL_USER_ROLES = `${URL_ADMIN}/users/roles`;
export const URL_USER_EDIT = `${URL_ADMIN}/users/edit`;
export const URL_TEAMS = `${URL_ADMIN}/teams`;
export const URL_TEAM_ASSIGNMENT = `${URL_TEAMS}/team-assignment`;
export const URL_TEAMS_EDIT = `${URL_TEAMS}/edit`;
export const URL_TEAMS_ADD = `${URL_TEAMS}/add`;
export const URL_BACKEND_LOGIN = '/fe/login';
export const URL_BACKEND_CALLBACK = '/fe/oauth/callback/opensrp';
export const URL_LOCATION_UNIT = `${URL_ADMIN}/location/unit`;
export const URL_LOCATION_UNIT_ADD = `${URL_ADMIN}/location/unit/add`;
export const URL_LOCATION_UNIT_EDIT = `${URL_ADMIN}/location/unit/edit/:id`;
export const URL_LOCATION_UNIT_GROUP = `${URL_ADMIN}/location/group`;
export const URL_LOCATION_UNIT_GROUP_ADD = `${URL_ADMIN}/location/group/add`;
export const URL_LOCATION_UNIT_GROUP_EDIT = `${URL_ADMIN}/location/group/edit/:id`;
export const URL_UPLOAD_JSON_VALIDATOR = `${URL_ADMIN}/form-config/json-validators/upload`;
export const URL_JSON_VALIDATOR_LIST = `${URL_ADMIN}/form-config/json-validators`;
export const URL_DRAFT_FILE_LIST = `${URL_ADMIN}/form-config/drafts`;
export const URL_UPLOAD_DRAFT_FILE = `${URL_ADMIN}/form-config/drafts/upload`;
export const URL_MANIFEST_RELEASE_LIST = `${URL_ADMIN}/form-config/releases`;
export const URL_DOWNLOAD_CLIENT_DATA = '/card-support/download-client-data';
