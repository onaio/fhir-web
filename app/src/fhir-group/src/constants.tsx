// url
export const URL_ADD_GROUP = 'group/add/';
export const URL_EDIT_GROUP = 'group/edit/';

//
// Groups
//
export const GROUP_UUID = 'group-uuid';
export const GROUP_GET = 'Group/';
export const GROUP_PUT = 'Group/';
export const GROUP_POST = 'Group/';

//
// Patient
//
export const PATIENT_GET = 'Patient/';