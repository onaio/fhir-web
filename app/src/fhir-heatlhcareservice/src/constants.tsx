// url
export const URL_ADD_HEALTHCARE = 'healthcare/add/';
export const URL_EDIT_HEALTHCARE = 'healthcare/edit/';

//
// Healthcares
//
export const HEALTHCARE_UUID = 'healthcare-uuid';
export const HEALTHCARES_GET = 'Group/';
export const HEALTHCARES_PUT = 'Group/';
export const HEALTHCARES_POST = 'Group/';
