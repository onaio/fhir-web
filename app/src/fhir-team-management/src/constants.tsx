// url
export const URL_ADMIN = 'admin';
export const URL_LIST_TEAMS = `${URL_ADMIN}/teams`;
export const URL_ADD_TEAM = 'teams/add/';
export const URL_EDIT_TEAM = 'teams/edit/';

//
// Practitioner
//
export const PRACTITIONERROLE_GET = 'PractitionerRole/';
export const PRACTITIONERROLE_POST = 'PractitionerRole/add/';
export const PRACTITIONERROLE_DEL = 'PractitionerRole/deleteByPractitioner/';

//
// Practitioner Role
//
export const PRACTITIONER_GET = 'Practitioner/';
export const PRACTITIONER_POST = 'Practitioner/';
export const PRACTITIONER_DEL = 'Practitioner/';

//
// Teams
//
export const TEAM_UUID = 'team-uuid';
export const TEAMS_GET = 'Organization/';
export const TEAMS_PUT = 'Organization/';
export const TEAMS_POST = 'Organization/';
