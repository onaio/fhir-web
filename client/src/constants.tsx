export const LOGOUT_REDIRECTION_DELAY = 1000;
export const REACT_CALLBACK_PATH = '/oauth/callback/:id';
export const LOGIN_PROMPT = 'Please log in with one of the following providers';
export const BACKEND_CALLBACK_PATH = '/fe/oauth/callback/:id';
export const USER_MANAGEMENT = 'User Management';
export const TEAMS = 'Teams';
export const LOCATIONS_UNIT = 'Locations unit';
export const LOCATIONS_UNIT_GROUP = 'Locations unit group';
export const LOCATIONS_UNIT_GROUP_SET = 'Locations unit group set';
export const PRODUCT_CATALOGUE = 'Product Catalogue';

// URLs
export const URL_EXPRESS_LOGIN = '/login';
export const URL_REACT_LOGIN = '/login';
export const URL_LOGOUT = '/logout';
export const URL_HOME = '/';
export const URL_ADMIN = '/admin';
export const URL_USER_EDIT = '/user/edit';
export const URL_TEAMS = '/teams';
export const URL_BACKEND_LOGIN = '/fe/login';
export const URL_BACKEND_CALLBACK = '/fe/oauth/callback/opensrp';
