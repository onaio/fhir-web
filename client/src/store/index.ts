// going to use this store initializer instead of the one in
// @opensrp/store package, for some reason its not working
// wehn passed to the redux provider
import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';
import {
  reducerName as keycloakUsersReducerName,
  reducer as keycloakUsersReducer,
} from '@opensrp/store';
import { Reducer } from 'redux';
import { RouterState } from 'connected-react-router';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
  router: connectReducer as Reducer<RouterState>,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;
// added this segment because registering from the user management
// package component is failing
defaultReducers[keycloakUsersReducerName] = keycloakUsersReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener((reducers) => {
  store.replaceReducer(combine(reducers));
});

export default store;
