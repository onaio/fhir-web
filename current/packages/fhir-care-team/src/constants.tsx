// url

// Query patams
export const SEARCH_QUERY_PARAM = 'searchQuery';

// FHIR API strings
export const careTeamResourceType = 'CareTeam';
export const groupResourceType = 'Group';
export const practitionerResourceType = 'Practitioner';
export const organizationResourceType = 'Organization';
export const URL_ADMIN = '/admin';
export const FHIR_CARE_TEAM = 'CareTeam';
export const FHIR_GROUPS = 'Group';
export const FHIR_PRACTITIONERS = 'Practitioner';
export const URL_EDIT_CARE_TEAM = `${URL_ADMIN}/CareTeam/edit`;
export const URL_CREATE_CARE_TEAM = `${URL_ADMIN}/CareTeam/new`;
export const URL_CARE_TEAM = `${URL_ADMIN}/CareTeams`;

// Route params
export const ROUTE_PARAM_CARE_TEAM_ID = 'careTeamId';

// form magic strings
export const uuid = 'uuid' as const;
export const id = 'id' as const;
export const name = 'name' as const;
export const status = 'status' as const;
export const practitionerParticipants = 'practitionerParticipants' as const;
export const organizationParticipants = 'organizationParticipants' as const;
export const managingOrganizations = 'managingOrganizations' as const;
