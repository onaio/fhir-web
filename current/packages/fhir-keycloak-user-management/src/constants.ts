export const practitionerResourceType = 'Practitioner';
export const careTeamResourceType = 'CareTeam';
export const organizationResourceType = 'Organization';
export const groupResourceType = 'Group';
export const practitionerRoleResourceType = 'PractitionerRole';

// keycloak endpoints strings
export const keycloakCountEndpoint = 'count';
