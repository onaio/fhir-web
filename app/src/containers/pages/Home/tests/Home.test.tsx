import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { history } from '@onaio/connected-reducer-registry';
import React from 'react';
import { Helmet } from 'react-helmet';
import { Router } from 'react-router';
import ConnectedHomeComponent, { Home, HomeProps } from '../Home';
import { getOpenSRPUserInfo } from '@onaio/gatekeeper';
import { store } from '@opensrp/store';
import { authenticateUser } from '@onaio/session-reducer';
import { Provider } from 'react-redux';
import { URL_LOCATION_UNIT, URL_LOCATION_UNIT_GROUP, URL_TEAMS } from '../../../../constants';

jest.mock('../../../../configs/env');

describe('containers/pages/Home', () => {
  it('renders without crashing', () => {
    const props = {
      extraData: {
        roles: ['ROLE_EDIT_KEYCLOAK_USERS'],
      },
    };
    shallow(<Home {...props} />);
  });

  it('renders Home correctly & changes Title of page', () => {
    const props = {
      extraData: {
        roles: ['ROLE_EDIT_KEYCLOAK_USERS'],
      },
    };
    const wrapper = mount(
      <Router history={history}>
        <Home {...props} />
      </Router>
    );

    const helmet = Helmet.peek();
    expect(helmet.title).toEqual('OpenSRP Web');
    expect(toJson(wrapper.find('Home'))).toMatchSnapshot('Home page rendered');
    wrapper.unmount();
  });

  it('works correctly with store', () => {
    const { authenticated, user, extraData } = getOpenSRPUserInfo({
      oAuth2Data: {
        access_token: 'hunter2',
        expires_in: '3599',
        state: 'opensrp',
        token_expires_at: '2017-07-13T20:30:59.000Z',
        token_type: 'bearer',
      },
      preferredName: 'Superset User',
      roles: ['ROLE_EDIT_KEYCLOAK_USERS'],
      username: 'superset-user',
    });
    store.dispatch(authenticateUser(authenticated, user, extraData));
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedHomeComponent />
        </Router>
      </Provider>
    );
    const connectedProps = wrapper.find('Home').props();
    expect((connectedProps as HomeProps).extraData).toEqual({
      oAuth2Data: {
        access_token: 'hunter2',
        expires_in: '3599',
        state: 'opensrp',
        token_expires_at: '2017-07-13T20:30:59.000Z',
        token_type: 'bearer',
      },
      preferredName: 'Superset User',
      roles: ['ROLE_EDIT_KEYCLOAK_USERS'],
      username: 'superset-user',
    });
  });

  it('displays links for location unit module', () => {
    const envModule = require('../../../../configs/env');
    envModule.ENABLE_LOCATIONS = true;

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedHomeComponent />
        </Router>
      </Provider>
    );

    expect(wrapper.find(`Link[to="${URL_LOCATION_UNIT}"]`)).toHaveLength(1);
    expect(wrapper.find(`Link[to="${URL_LOCATION_UNIT_GROUP}"]`)).toHaveLength(1);
  });

  it('displays links for teams module', () => {
    const envModule = require('../../../../configs/env');
    envModule.ENABLE_TEAMS = true;

    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedHomeComponent />
        </Router>
      </Provider>
    );

    expect(wrapper.find(`Link[to="${URL_TEAMS}"]`)).toHaveLength(1);
  });
});
