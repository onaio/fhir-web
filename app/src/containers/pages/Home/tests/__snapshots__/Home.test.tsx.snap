// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/Home renders Home correctly & changes Title of page: Home page rendered 1`] = `
<Home
  extraData={
    Object {
      "roles": Array [
        "ROLE_EDIT_KEYCLOAK_USERS",
      ],
    }
  }
>
  <div
    className="text-center home"
  >
    <HelmetWrapper
      defer={true}
      encodeSpecialCharacters={true}
    >
      <SideEffect(NullComponent)
        defer={true}
        encodeSpecialCharacters={true}
        title="OpenSRP Web"
        titleAttributes={Object {}}
      >
        <NullComponent
          defer={true}
          encodeSpecialCharacters={true}
          title="OpenSRP Web"
          titleAttributes={Object {}}
        />
      </SideEffect(NullComponent)>
    </HelmetWrapper>
    <Row
      className="weclome-box"
      justify="center"
    >
      <div
        className="ant-row ant-row-center weclome-box"
        style={Object {}}
      >
        <Col
          span={6}
        >
          <div
            className="ant-col ant-col-6"
            style={Object {}}
          >
            <h3>
              Welcome to OpenSRP
            </h3>
          </div>
        </Col>
      </div>
    </Row>
    <Row
      className="links-box"
      gutter={16}
    >
      <div
        className="ant-row links-box"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
          }
        }
      >
        <Col
          className="gutter-row"
          span={12}
        >
          <div
            className="ant-col ant-col-12 gutter-row"
            style={
              Object {
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <Link
              className="admin-link"
              to="/admin/users"
            >
              <LinkAnchor
                className="admin-link"
                href="/admin/users"
                navigate={[Function]}
              >
                <a
                  className="admin-link"
                  href="/admin/users"
                  onClick={[Function]}
                >
                  <Button
                    block={false}
                    className="btn-links"
                    color="outline"
                    ghost={false}
                    htmlType="button"
                    loading={false}
                  >
                    <Wave>
                      <button
                        className="ant-btn btn-links"
                        color="outline"
                        onClick={[Function]}
                        type="button"
                      >
                        <LoadingIcon
                          existIcon={false}
                          loading={false}
                          prefixCls="ant-btn"
                        >
                          <CSSMotion
                            motionName="ant-btn-loading-icon-motion"
                            onAppearActive={[Function]}
                            onAppearStart={[Function]}
                            onEnterActive={[Function]}
                            onEnterStart={[Function]}
                            onLeaveActive={[Function]}
                            onLeaveStart={[Function]}
                            removeOnLeave={true}
                            visible={false}
                          >
                            <DomWrapper />
                          </CSSMotion>
                        </LoadingIcon>
                        <span
                          key=".0"
                        >
                          Admin
                        </span>
                      </button>
                    </Wave>
                  </Button>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </Col>
      </div>
    </Row>
    <Row
      className="links-box"
      gutter={16}
    >
      <div
        className="ant-row links-box"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
          }
        }
      />
    </Row>
    <Row
      className="links-box"
      gutter={16}
    >
      <div
        className="ant-row links-box"
        style={
          Object {
            "marginLeft": -8,
            "marginRight": -8,
          }
        }
      />
    </Row>
  </div>
</Home>
`;
